src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -304,9 +329,7 @@ CHIP_ERROR ChipDeviceScanner::MainLoopStartScan(ChipDeviceScanner * self)
     if (!bluez_adapter1_call_start_discovery_sync(self->mAdapter, self->mCancellable, &MakeUniquePointerReceiver(error).Get()))
     {
         ChipLogError(Ble, "Failed to start discovery: %s", error->message);
-
-        self->mIsScanning = false;
-        self->mDelegate->OnScanComplete();
+        return CHIP_ERROR_INTERNAL;
```

Comment:
Don't you need to reset the state to initialized in this case so that subsequent scans work?

Reply:
Ah, I see `GLibMatterContextInvokeSync` returns the error of the called function. Thanks for explaining that.